backend/src/main/kotlin/app/ehrenamtskarte/backend/matomo/Matomo.kt

```diff
@@ -0,0 +1,159 @@
+package app.ehrenamtskarte.backend.matomo
+
+import app.ehrenamtskarte.backend.config.MatomoConfig
+import app.ehrenamtskarte.backend.config.ProjectConfig
+import app.ehrenamtskarte.backend.stores.webservice.schema.SearchParams
+import app.ehrenamtskarte.backend.verification.database.CodeType
+import app.ehrenamtskarte.backend.verification.database.repos.CardRepository
+import app.ehrenamtskarte.backend.verification.webservice.schema.types.CardGenerationModel
+import jakarta.servlet.http.HttpServletRequest
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.matomo.java.tracking.MatomoRequest
+import org.matomo.java.tracking.MatomoRequestBuilder
+import org.matomo.java.tracking.MatomoTracker
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.util.concurrent.ExecutionException
+
+enum class EventCategory {
+    Cards
+}
+
+object Matomo {
+    val logger = LoggerFactory.getLogger(Matomo::class.java)
+    private const val ACTION_NAME = "Request"
+
+    private fun sendTrackingRequest(matomoConfig: MatomoConfig, requestBuilder: MatomoRequestBuilder) {
+        val siteId = matomoConfig.siteId
+        val url = matomoConfig.url
+        val tracker = MatomoTracker(url)
+
+        val matomoRequest = requestBuilder
+            .actionName(ACTION_NAME)
+            .siteId(siteId)
+            .authToken(matomoConfig.accessToken)
+            .build()
+
+        try {
+            val response = tracker.sendRequestAsync(matomoRequest)
+            val httpResponse = response.get()
```

Comment:
I feel like, we should not await this future here, to not unnecessarily increase the delay the response of our API.
We should do the error logging asynchronously too.